### PR TITLE
Updated react query to v4

### PR DESCRIPTION
**Pull request by tawandachiteshe**

### CODE REVIEW

**Comment by rametta** [comment]
I'm not sure we're ready to upgrade to v4 yet. Is this something that is needed for you @tawandachiteshe ?

**Comment by tawandachiteshe** [comment]
Its not that important i have to correct the generated file everytime i change my openapi file. But its okay i will wait for a version change.

File: src/rapini-mutation.ts [hunk 1]
```diff
@@ -309,7 +309,7 @@ export function makeRapiniMutation() {
                             ts.factory.createIdentifier("context"),
                             ts.factory.createToken(ts.SyntaxKind.QuestionToken),
                             ts.factory.createTypeReferenceNode(
-                              ts.factory.createIdentifier("TContext"),
+                              ts.factory.createIdentifier("TContext | undefined"),
```
Comment (rametta) [inline src/rapini-mutation.ts:312]: The question token on line 310 covers the undefined type, this shouldn't be needed

Comment (tawandachiteshe) [reply src/rapini-mutation.ts:312]: So in v4 it was giving an error saying context doesn't match type undefined it made the error go away. Coz when you generate it the type isn't there. As Context | undefined.

Comment (rametta) [reply src/rapini-mutation.ts:312]: ah okay so it's a v4 thing. Either way, We shouldn't do it like this. Should create a proper ast union type node like:
```ts
ts.factory.createUnionTypeNode([
  ts.factory.createTypeReferenceNode(ts.factory.createIdentifier("TContext")),
  ts.factory.createKeywordTypeNode(ts.SyntaxKind.UndefinedKeyword)
])
```

Comment (tawandachiteshe) [reply src/rapini-mutation.ts:312]: Ooo i used a hacky solution classic beginner mistake. Okay thanks for your time. I will close the pr.

**Comment by rametta** [comment]
I'll probably add a CLI flag for v4, like `--v4` to generate v4 version since I still want to support v3. @tawandachiteshe Would you like to implement that?

**Pull request closed without merging**